text/metrics/0238-pipeline-monitoring.md

```diff
@@ -0,0 +1,426 @@
+Because the practice of asynchronous reporting is widespread,
+OpenTelemetry Collectors therefore should normally count exported data
+in addition to received data, despite creating redundancy when errors
+are not suppressed.
+
+#### Pipeline stage-name uniqueness
+
+The Pipeline Stage Name Uniqueness requirement developed here avoids
+over-counting in an export pipeline by ensuring that no single metric
+name counts items more than once in transit.  This rule prevents
+counting items of telemetry sent by SDKs and Collectors in the same
+metric; it also prevents counting items of telemetry sent through a
+multi-tier arrangement of Collectors using the same metric.
+
+In a standard deployment of OpenTelemetry, we expect one, two, or
+three stations in a collection pipeline.  The names given to these
+standard set of stations:
+
+- `sdk`: an original source of new telemetry
+- `agent`: a collector with operations "local" to the `sdk`
+- `gateway`: a collector serving as a proxy to an external service.
+
+This is not meant as an exclusive set of station names.  Users should
+be given the ability to configure the station name used by particular
+instances of the OpenTelemetry Collector.  It may even be desirable to
+support configuring "sub-stations" within a larger pipeline, for
+example when there are connectors in use; however, if so, the
+collector must enforce that pipeline-stage names are unique within a
+station.
+
+#### Pipeline signal type
+
+OpenTelemetry currently has 3 signal types, but it may add more.
+Instead of using the signal name in the metric names, we opt for a
+general-purpose noun that usefully describes any signal.
+
+The signal-agnostic term used here is "items", referring to spans, log
+records, and metric data points.  An attribute to distinguish the
+`signal` will be used with name `traces`, `logs`, or `metrics`.
+
+Users are expected to understand that the data item for traces is a
+span, for logs is a record, and for metrics is a point.  Users may
+opt-in to removing this attribute, in which case items of telemetry
+data will be counted in aggregate.  When the `signal` attribute is
+removed, loss-rate can likewise only be calculated in aggregate.
+
+#### Pipeline component name
+
+Components are uniquely identified using a descriptive `name`
+attribute which encompasses at least a short name describing the type
+of component being used (e.g., `batch` for the SDK BatchSpanProcessor
+or the Collector batch proessor).
+
+When there is more than one component of a given type active in a
+pipeline having the same `domain` and `signal` attributes, the `name`
+should include additional information to disambiguate the multiple
+instances using the syntax `<type>/<instance>`.  For example, if there
+were two `batch` processors in a collection pipeline (e.g., one for
+error spans and one for non-error spans) they might use the names
+`batch/error` and `batch/noerror`.
```

Comment:
In the SDK, if two batch processors are present, they're only differentiable by the associated exporter. You can't configure the SDK to send some spans to one batch processor, and others to another. So maybe the name would have to be something like `batch/otlp`, `batch/zipkin`, etc.